### DISCUSSION TITLE
Inline validation in web forms

### DISCUSSION THREAD

**Discussion by limakid** [twbs/bootstrap#34959] [Q&A]
Is there a way I can add inline validation to a web form? Using the validation tools of Bootstrao 5, **The errors are shown only after submitted the form**. I‘d like to show the error to the user right after filled up the respective field with the error, an input text with a username for example.

So, if I admit only A-Z characters (already declared in the pattern), as long as he put a number or strange character, I‘d like to show the error he is in making so he can correct inmediatly or later before submitting, so at least he is warned already and will not wait to press “submit” to become aware that he was filling the form in the wrong way.

What I want is to keep the vaklidation classes given by bootstrap, I just want to add these features I mentioned. Is it possible or better I program all validation from zero? Of course I talk about client side. Ideas, please.

**Comment by Bhakti087** [comment]
Have you find any ans to this problem?

**Reply by gsudhesh** [reply]
@limakid you can call the validation on the text box change event  and the include the following class.

```
$(control).change(function(){
 if(invalid)
{
  $(control).removeClass("is-valid is-invalid").addClass("is-invalid");
}
else
{  
$(control).removeClass("is-valid is-invalid").addClass("is-valid");
}
//  set custom message dynamically
 $(control).closest("div").find("div.invalid-feedback").text(errormessage); 
});
```

```
  <form id ='myform' class="row g-3 needs-validation" novalidate>
    <div class="col-md-3 mb-4" >
    <input type="text" id="txtnumber" class="form-control" maxlength="11" required />
    <div class="invalid-feedback">Field is required</div>
    </div>
</form>
```
